doc/api/esm.md

```diff
@@ -299,6 +299,8 @@ The `type: 'json'` attribute is mandatory when importing JSON modules.
 
 ## Built-in modules
 
+> **Note:** When importing built-in modules via ESM , the module is fully evaluated at import time, and all properties are populated immediately. This can make initial imports slightly slower compared to CommonJS, where the module body is still evaluated eagerly but some built-in properties are only initialized when first accessed.
```

Comment:
```suggestion
> When importing built-in modules, all the named exports (i.e. properties of the module exports object) are populated even if they are not individually accessed.
> This can make initial imports slightly slower compared to loading them with `require()` or
> `process.getBuiltInModule()`, where the module exports object is evaluated immediately,
> but some of its properties may only be initialized when first accessed individually.
```

`import` is not a ESM-only thing - CommonJS also has access to `import()`. On the other hand, `require()` is not ESM-only either, ESM can use `createRequire()` too, and `process.getBuiltInModule()`.